pkg/piecestore/pstore.go

```diff
@@ -20,86 +19,70 @@ var (
+	folder2 := string(id[2:4])
+	fileName := string(id[4:])
 
 	return path.Join(dir, folder1, folder2, fileName), nil
 }
 
-// Store -- Store data into piece store
-// 	hash 					(string)		Hash of the data to be stored
-// 	r 			  		(io.Reader)	File/Stream that contains the contents of the data to be stored
-// 	length 				(length)		Size of the data to be stored
-// 	psFileOffset 	(offset)  	Offset of the data that you are writing. Useful for multiple connections to split the data transfer
-// 	dir 					(string)		pstore directory containing all other data stored
-// 	returns 			(error) 		error if failed and nil if successful
-func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) (int64, error) {
+// StoreWriter -- Store data into piece store in multiple writes
+// 	id 					(string)								Id of the data to be stored
+// 	dir 					(string)								pstore directory containing all other data stored
+// 	returns 			(*fpiece.Chunk, error) 	error if failed and nil if successful
+func StoreWriter(id string, length int64, psFileOffset int64, dir string) (*fpiece.Chunk, error) {
 	if psFileOffset < 0 {
-		return 0, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
-	}
-	if length < 0 {
-		return 0, ArgError.New("Length is less than 0. Must be greater than or equal to 0")
-	}
-	if dir == "" {
-		return 0, ArgError.New("No path provided")
+		return nil, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
 	}
 
-	dataPath, err := PathByHash(hash, dir)
+	dataPath, err := PathByID(id, dir)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	// Create directory path on file system
 	if err = os.MkdirAll(filepath.Dir(dataPath), 0700); err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	// Create File on file system
 	dataFile, err := os.OpenFile(dataPath, os.O_RDWR|os.O_CREATE, 0755)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	dataFileChunk, _ := fpiece.NewChunk(dataFile, psFileOffset, length)
-
-	// Close when finished
-	defer dataFile.Close()
-
-	return io.CopyN(dataFileChunk, r, length)
+	return fpiece.NewChunk(dataFile, psFileOffset, length)
 }
 
-// Retrieve -- Retrieve data from pstore directory
-//	hash 					(string)		   Hash of the stored data
-//	w 						(io.Writer)	   Stream that recieves the stored data
-//	length 				(length)		   Amount of data to read. Read all data if -1
-//	readPosOffset	(offset)	   	 Offset of the data that you are reading. Useful for multiple connections to split the data transfer
-//	dir 					(string)		   pstore directory containing all other data stored
-//	returns 			(int64, error) returns err if failed and the number of bytes retrieved if successful
-func Retrieve(hash string, w io.Writer, length int64, readPosOffset int64, dir string) (int64, error) {
-	if dir == "" {
-		return 0, ArgError.New("No path provided")
-	}
-
-	dataPath, err := PathByHash(hash, dir)
+// RetrieveReader -- Retrieve data from pstore directory
```

Comment:
same as above. I'll leave these comments off for the rest of the PR but it applies to the whole.